pkg/api/server/server.go

```diff
@@ -130,9 +139,19 @@ func NewDefault(serviceName, addr string, opts ...Option) *DefaultServer {
 	return s
 }
 
-// Router returns the gin.Engine used by the DefaultServer
-func (server *DefaultServer) Router() *gin.Engine {
-	return server.router
+// API returns the huma API server which is used to register and document endpoints
+func (server *DefaultServer) API() huma.API {
+	return server.api
+}
+
+// OpenAPI writes the full OpenAPI spec to the writer w
+func (server *DefaultServer) OpenAPI(w io.Writer) error {
+	b, err := server.api.OpenAPI().DowngradeYAML()
+	if err != nil {
+		return fmt.Errorf("failed to generate OpenAPI spec: %w", err)
+	}
+	_, err = w.Write(b)
+	return err
```

Comment:
I'd maybe wrap this in a more descriptive error (e.g. `failed to write OpenAPI spec to destination: %w`) - errors from io.Writers are usually not very helpful (`EOF` or whatever) and unspecific so it may be difficult to trace the error here...